### ISSUE TITLE
500 Internal Server Error for dex token verification

### ISSUE THREAD

**Issue by sreeramu** [eclipse/che#21645]
### Describe the bug

Getting 500 Internal Server Error for dex token verification

kubectl logs -n eclipse-che -l component=che-gateway -c oauth-proxy -f

[2022/08/21 07:27:08] [oauthproxy.go:723] Error redeeming code during OAuth2 callback: token exchange failed: Post "https://dex.192.168.49.2.nip.io/token": dial tcp 127.0.0.1:443: connect: connection refused
172.17.0.6:45530 - e27c929c37f4075a864067ecae1139eb - - [2022/08/21 07:27:08] 192.168.49.2.nip.io GET - "/oauth/callback?code=onq6r5kow3i2nyayovhfeei5e&state=q8gAXewnUErKN8rbRiaBiTKjVKpelL8aDUXImDempAc%3A%2Fdashboard" HTTP/1.1 "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36" 500 2830 0.045
[2022/08/21 07:27:08] [oauthproxy.go:862] No valid authentication in request. Initiating login.
172.17.0.6:45530 - f13afe88e82d9f5a9fdf97642024517a - - [2022/08/21 07:27:08] 192.168.49.2.nip.io GET - "/favicon.ico" HTTP/1.1 "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36" 302 302 0.000

### Che version

7.51@latest

### Steps to reproduce

1) Open che dashbord url 
2) Enter username and password
3) next page 500 error

### Expected behavior

Should login success and take to dashboard home page

### Runtime

minikube

### Screenshots

<img width="708" alt="Screenshot 2022-08-21 at 1 32 07 PM" src="https://user-images.githubusercontent.com/438259/185781701-f804dd66-cb0f-4364-a479-99c904d618db.png">

### Installation method

chectl/latest

### Environment

macOS

### Eclipse Che Logs

```shell
kubectl logs -n eclipse-che -l component=che-gateway -c oauth-proxy -f

[2022/08/21 07:27:08] [oauthproxy.go:723] Error redeeming code during OAuth2 callback: token exchange failed: Post "https://dex.192.168.49.2.nip.io/token": dial tcp 127.0.0.1:443: connect: connection refused
172.17.0.6:45530 - e27c929c37f4075a864067ecae1139eb - - [2022/08/21 07:27:08] 192.168.49.2.nip.io GET - "/oauth/callback?code=onq6r5kow3i2nyayovhfeei5e&state=q8gAXewnUErKN8rbRiaBiTKjVKpelL8aDUXImDempAc%3A%2Fdashboard" HTTP/1.1 "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36" 500 2830 0.045
[2022/08/21 07:27:08] [oauthproxy.go:862] No valid authentication in request. Initiating login.
172.17.0.6:45530 - f13afe88e82d9f5a9fdf97642024517a - - [2022/08/21 07:27:08] 192.168.49.2.nip.io GET - "/favicon.ico" HTTP/1.1 "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/104.0.0.0 Safari/537.36" 302 302 0.000
```

**Comment by tolusha** [comment]
@sreeramu 
Could you check as well che-server container logs?
`kubectl logs -n eclipse-che -l component=che`

**Comment by sreeramu** [comment]
@tolusha no new log on preforming this operation.

sreeramu@Sreenivasans-MacBook-Pro ~ % kubectl logs -n eclipse-che -l component=che -f
2022-08-20 08:04:43,932[main]             [INFO ] [o.e.c.m.oidc.OIDCInfoProvider 72]    - Retrieving OpenId configuration from endpoint: http://dex.dex:5556/.well-known/openid-configuration
2022-08-20 08:04:44,168[main]             [INFO ] [o.e.c.m.oidc.OIDCInfoProvider 81]    - openid configuration = {issuer=https://dex.192.168.49.2.nip.io, authorization_endpoint=https://dex.192.168.49.2.nip.io/auth, token_endpoint=https://dex.192.168.49.2.nip.io/token, jwks_uri=https://dex.192.168.49.2.nip.io/keys, userinfo_endpoint=https://dex.192.168.49.2.nip.io/userinfo, device_authorization_endpoint=https://dex.192.168.49.2.nip.io/device/code, grant_types_supported=[authorization_code, refresh_token, urn:ietf:params:oauth:grant-type:device_code], response_types_supported=[code], subject_types_supported=[public], id_token_signing_alg_values_supported=[RS256], code_challenge_methods_supported=[S256, plain], scopes_supported=[openid, email, groups, profile, offline_access], token_endpoint_auth_methods_supported=[client_secret_basic, client_secret_post], claims_supported=[iss, sub, aud, iat, exp, email, email_verified, locale, name, preferred_username, at_hash]}
2022-08-20 08:04:44,386[main]             [INFO ] [o.e.c.a.c.u.ApiInfoLogInformer 36]   - Eclipse Che Api Core: Build info '7.52.0' scmRevision 'c3b9cbd59eac5ac328d5638c53c07dfa1182277e' implementationVersion '7.52.0'
2022-08-20 08:04:44,464[main]             [WARN ] [p.s.AdminPermissionInitializer 69]   - Admin admin not found yet.
2022-08-20 08:04:44,542[main]             [INFO ] [o.e.c.c.metrics.MetricsServer 46]    - Metrics server started at port 8087 successfully 
20-Aug-2022 08:04:45.550 INFO [main] org.apache.catalina.startup.HostConfig.deployWAR Deployment of web application archive [/home/user/eclipse-che/tomcat/webapps/api.war] has finished in [52,055] ms
20-Aug-2022 08:04:45.553 INFO [main] org.apache.catalina.startup.HostConfig.deployWAR Deploying web application archive [/home/user/eclipse-che/tomcat/webapps/ROOT.war]
20-Aug-2022 08:04:49.333 INFO [main] org.apache.catalina.startup.HostConfig.deployWAR Deployment of web application archive [/home/user/eclipse-che/tomcat/webapps/ROOT.war] has finished in [3,781] ms
20-Aug-2022 08:04:49.351 INFO [main] org.apache.coyote.AbstractProtocol.start Starting ProtocolHandler ["http-nio-8080"]
20-Aug-2022 08:04:49.365 INFO [main] org.apache.catalina.startup.Catalina.start Server startup in [57909] milliseconds

**Comment by tolusha** [comment]
It is hard to say what is wrong there.
I don't like this error
```
[2022/08/21 07:27:08] [oauthproxy.go:723] Error redeeming code during OAuth2 callback: token exchange failed: Post "https://dex.192.168.49.2.nip.io/token": dial tcp 127.0.0.1:443: connect: connection refused
```
Could you inspect dex container logs as well?

**Comment by sreeramu** [comment]
@tolusha 
I will tell what could be the issue , you will be able to find is it correct or not.
I had installed same setup in two systems Mac & ubuntu

In Mac from the gateway pod system if i try to ping dex.192.168.49.2.nip.io it resolves as 127.0.0.1
In ubuntu if i try to ping dex.192.168.49.2.nip.io it was not reachable.

So for testing i had added the mapping for dex.192.168.49.2.nip.io to the ingress ip in /etc/hosts after adding this the 500 error is solved.

looks like it is host resolving issue.
